common/src/main/java/org/cloudfoundry/identity/uaa/authorization/UaaAuthorizationEndpoint.java

```diff
@@ -418,24 +430,20 @@ private String getUnsuccessfulRedirect(AuthorizationRequest authorizationRequest
         }
 
         UriComponentsBuilder template = UriComponentsBuilder.fromUriString(authorizationRequest.getRedirectUri());
-        Map<String, String> query = new LinkedHashMap<String, String>();
```

Comment:
Based on the assumption that the URI is encoded, we can't expand variables. In order to expand variables in UriComponentsBuilder, the URI must be decoded.